### PR TITLE
attempt to enhance message types for rx-jupyter

### PR DESCRIPTION
**Pull request by rgbkrk**
Bit of a follow up to https://github.com/nteract/nteract/pull/3571.

Since I'm just getting started on adapting to typescript, I'd love any feedback.

### CODE REVIEW

File: packages/rx-jupyter/src/kernels.ts [hunk 1]
```diff
@@ -114,12 +114,17 @@ export const formWebSocketURL = (
   return url.replace(/^http(s)?/, "ws$1");
 };
 
+// TODO: Figure out how/where to bring over `@nteract/messaging` types for messages into `rx-jupyter`
```
Comment (captainsafia) [inline packages/rx-jupyter/src/kernels.ts]:
```suggestion
import { JupyterMessage } from '@nteract/messaging';
```

File: packages/rx-jupyter/src/kernels.ts [hunk 2]
```diff
@@ -114,12 +114,17 @@ export const formWebSocketURL = (
...
+  header: {};
+};
+
 export const connect = (
   serverConfig: ServerConfig,
   kernelID: string,
   sessionID?: string
 ) => {
-  const wsSubject = webSocket<string>(
+  const wsSubject = webSocket<Message>(
```
Comment (captainsafia) [inline packages/rx-jupyter/src/kernels.ts]:
```suggestion
  const wsSubject = webSocket<JupyterMessage>(
```

File: packages/rx-jupyter/src/kernels.ts [hunk 3]
```diff
@@ -156,7 +161,7 @@ export const connect = (
   // and ensuring it's serialized
   return Subject.create(
     Subscriber.create(
-      (message: any) => {
+      (message?: Message) => {
```
Comment (captainsafia) [inline packages/rx-jupyter/src/kernels.ts]:
```suggestion
      (message?: JupyterMessage) => {
```

**Review by captainsafia — COMMENTED** [review]
Posted some suggestions on how I think it might look.

I think we might be able to use the generics in the `JupyterMessage` interface to constrain the particular message we expect to get from the `connect`.

File: packages/rx-jupyter/src/kernels.ts (hunk 1 above)
Comment (rgbkrk) [reply packages/rx-jupyter/src/kernels.ts]: I love that we can import typescript types without an import type syntax.

**Comment by rgbkrk** [comment]
I think I'll rebase since I left the default message when using the "suggested changes" feature.

File: packages/rx-jupyter/src/kernels.ts [hunk 4]
```diff
@@ -114,12 +114,17 @@ export const formWebSocketURL = (
   return url.replace(/^http(s)?/, "ws$1");
 };
 
+import { JupyterMessage } from '@nteract/messaging';
+type Message = {
```
Comment (jdetle) [inline packages/rx-jupyter/src/kernels.ts]: I dont think this is getting used anywhere.

**Comment by rgbkrk** [comment]
This works really well now, thank you @captainsafia.

**Pull request merged**